tests/test_directives.py

```diff
@@ -145,8 +170,8 @@ def test_set_defined_directive(self, setUp, available_directives_list):
 
     def test_set_defined_directive_invalid(self, setUp, available_directives_list):
```

Comment:
This tests one case, see comment above on `test_none_default_value`. A test that looks to see if many different invalid values raise Errors would be good.

Reply:
I have tested this is the `TestItems` class. If you want, I can write here too.